app/rdf/query-cube-preview.ts

```diff
@@ -15,6 +16,12 @@ import * as ns from "./namespace";
 import { getDataKind, getScaleType, timeFormats, timeUnits } from "./parse";
 import { buildLocalizedSubQuery } from "./query-utils";
 
+type ShortQuad = Omit<Quad, "subject" | "predicate" | "object"> & {
+  s: Quad["subject"];
+  p: Quad["predicate"];
+  o: Quad["object"];
+};
```

Comment:
I am bit surprised that it's necessary to have this while we can destructure and rename at the levels of functions, I wonder if by using in/out on the quads, we would not manage to write shorter code with no need for ShortQuad.

Reply:
That was done for convenience and to force the same naming across several places we used the `Quad`s – it could be done in each individual map, but to use `.out` we would need to completely change the structure of the logic 🥲